src/com/google/enterprise/adaptor/fs/FsAdaptor.java

```diff
@@ -668,14 +683,13 @@ private Path getDfsRoot(Path path) throws IOException {
 
   /** Verify the path is available and we have access to it. */
   private void validateShare(Path sharePath) throws IOException {
-    if (delegate.isDfsNamespace(sharePath)) {
-      throw new AssertionError("validateShare may only be called "
-          + "on DFS links or active storage paths.");
-    }
-
     // Verify that the adaptor has permission to read the contents of the root.
     try {
-      delegate.newDirectoryStream(sharePath).close();
+      if (delegate.isDfsNamespace(sharePath) && !allowFilesInDfsNamespaces) {
+        delegate.enumerateDfsLinks(sharePath);
```

Comment:
This is going to happen later (in the code above), whether allowFiles is true or false. So why do this here now with the added condition?